Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardInput/CreditCardInputViewModel.swift

```diff
@@ -0,0 +1,95 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+class CreditCardInputViewModel: ObservableObject {
+    let fieldType: CreditCardInputType
+    let fieldHeadline: String
+    let errorString: String
+    var formattedTextLimit: Int
+    var delimiterCharacter: String?
+    var keyboardType: UIKeyboardType
+
+    @Published var isValid = true
+
+    init(fieldType: CreditCardInputType) {
+        self.fieldType = fieldType
+
+        switch fieldType {
+        case .name:
+            self.fieldHeadline = .CreditCard.EditCard.NameOnCardTitle
+            self.errorString = .CreditCard.ErrorState.NameOnCardSublabel
+            self.delimiterCharacter = nil
+            self.formattedTextLimit = 100 // TBA
+            self.keyboardType = .alphabet
+        case .number:
+            self.fieldHeadline = .CreditCard.EditCard.CardNumberTitle
+            self.errorString = .CreditCard.ErrorState.CardNumberSublabel
+            self.formattedTextLimit = 20
+            self.delimiterCharacter = " "
+            self.keyboardType = .numberPad
+        case .expiration:
+            self.fieldHeadline = .CreditCard.EditCard.CardExpirationDateTitle
+            self.errorString = .CreditCard.ErrorState.CardExpirationDateSublabel
+            self.formattedTextLimit = 7
+            self.delimiterCharacter = " / "
+            self.keyboardType = .numberPad
+        case .securityCode: // Design feedback needed
+            self.fieldHeadline = "Security Code"
+            self.errorString = "Enter the three digit security code"
+            self.formattedTextLimit = 3
+            self.delimiterCharacter = nil
+            self.keyboardType = .numberPad
+        }
+    }
+
+    func updateNameValidity(userInputtedText: String) {
+        isValid = !userInputtedText.isEmpty
+    }
+
+    func updateCardNumberValidity(userInputtedText: String) {
+        isValid = userInputtedText.count >= 15 && userInputtedText.count <= 16
+    }
+
+    func updateExpirationValidity(userInputtedText: String) {
```

Comment:
Working with dates is hard we should add more test for edge cases like if the user enters letter or invalid month etc

Reply:
The interface is expecting a string so it could be misused. Having the test will protect the code for future changes where user input as String can be wrongly use. Also what happens with external keyboards someone can probably enter something else than a number?

Reply:
Got it okay! Will add some tests for that case too. Thanks for pointing it out